lib/lbt/bundle/Builder.js

```diff
@@ -471,123 +461,77 @@ class BundleBuilder {
 	}
 }
 
-const CALL_DEFINE = ["define"];
 const CALL_SAP_UI_DEFINE = ["sap", "ui", "define"];
 
-function rewriteDefine(targetBundleFormat, code, moduleName, avoidLazyParsing) {
-	function _injectModuleNameIfNeeded(defineCall) {
-		if ( defineCall.arguments.length == 0 ||
-				defineCall.arguments[0].type !== Syntax.Literal ) {
-			defineCall.arguments.unshift({
-				type: Syntax.Literal,
-				value: ModuleName.toRequireJSName(moduleName)
-			});
-		}
-	}
-
-	function enforceEagerParsing(defineCall) {
-		const args = defineCall.arguments;
-		// wrap factory function to avoid lazy parsing (V8 and Chakra)
-		let iFactory = 0;
-		// skip any hard coded module name
-		if ( iFactory < args.length && args[iFactory].type === Syntax.Literal ) {
-			iFactory++;
-		}
-		// skip array of dependencies
-		if ( iFactory < args.length && args[iFactory].type === Syntax.ArrayExpression ) {
-			iFactory++;
-		}
-		// wrap factory
-		if ( iFactory < args.length && args[iFactory].type === Syntax.FunctionExpression ) {
-			/* NODE-TODO
-			Token firstToken = doc.getToken(args.getChild(iFactory).getTokenStartIndex());
-			Token lastToken = doc.getToken(args.getChild(iFactory).getTokenStopIndex());
-			if ( "function".equals(firstToken.getText())
-					 && "}".equals(lastToken.getText()) ) {
-				firstToken.setText("(" + firstToken.getText());
-				lastToken.setText(lastToken.getText() + ")");
-			} */
-		}
-	}
-
+function rewriteDefine(targetBundleFormat, code, moduleName) {
 	let ast;
+	const codeStr = code.toString();
 	try {
-		ast = esprima.parseScript(code.toString(), {loc: true});
+		ast = esprima.parseScript(codeStr, {range: true});
 	} catch (e) {
 		log.error("error while parsing %s: %s", moduleName, e.message);
 		return;
 	}
 
 	if ( ast.type === Syntax.Program &&
-			ast.body.length === 1 &&
-			ast.body[0].type === Syntax.ExpressionStatement ) {
-		// rewrite define to require.predefine
-		if ( targetBundleFormat.supportsNativeDefine() && isMethodCall(ast.body[0].expression, CALL_DEFINE) ) {
```

Comment:
Has the `targetBundleFormat.supportsNativeDefine()` check be removed on purpose?
If so and in case we don't expect them to be used somewhere else I'd suggest to also remove the functions from the bundle formats. Currently both formats just return `false`.

Reply:
After reading the previous commit messages, I guess I can answer the first question myself with a "Yes" 😄

Reply:
Done 👍